### PR TITLE
FPO-290

### PR DESCRIPTION
**Pull request by Lee365**
### Jira link
https://transformuk.atlassian.net/browse/FPO-290

### What?

I have added/removed/altered:

- [x] Added
- [x] Removed
- [x] Altered

### Why?

I am doing this because:

-

### AC

- Tested
- Shared with stakeholders
- Pair reviewed

### CODE REVIEW

File: src/controllers/apiKeysController.ts [hunk 1]
```diff
@@ -31,12 +31,18 @@ export class ApiKeyController {
     const organisationId = req.params.organisationId
     const description = req.body.description
     const userId = req.headers['x-user-id'] ?? ''
+    const applicationReference = req.body.applicationReference
```
Comment (ashward) [inline src/controllers/apiKeysController.ts:34]: What do we need the `applicationReference` for when creating a new API key? Doesn't seem to be used anywhere.

File: spec/controllers/apiKeysControllerSpec.ts [hunk 2]
```diff
@@ -230,36 +230,6 @@ describe('ApiKeyController', () => {
         params: { organisationId: 'organisationId' },
         body: { description: 'description' }
       } as any
-      const res = {
```
Comment (ashward) [inline spec/controllers/apiKeysControllerSpec.ts:233]: Any reason this chunk of the test has been deleted?

Comment (Lee365) [reply spec/controllers/apiKeysControllerSpec.ts:233]: Mmm . not sure. Added this back.

File: src/controllers/apiKeysController.ts (hunk 1 above)
Comment (Lee365) [reply src/controllers/apiKeysController.ts:34]: Removed

File: src/operations/updateOrganisation.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import { UpdateItemCommand, type UpdateItemCommandInput, type DynamoDBClient } from '@aws-sdk/client-dynamodb'
+
+const TableName = process.env.ORGANISATIONS_TABLE_NAME ?? ''
```
Comment (ashward) [inline src/operations/updateOrganisation.ts:3]: Shouldn't we default this? Seems a bit much to require such a basic thing to be configured?

Comment (ashward) [reply src/operations/updateOrganisation.ts:3]: Also variable/const names should be camelCase

File: src/routes/api.ts [hunk 4]
```diff
@@ -48,6 +48,8 @@ router.delete('/keys/:organisationId/:id', (req, res) => { apiKeyController.dest
 router.post('/users/:id', (req, res) => { userController.create(req, res) })
 router.get('/users/:id', (req, res) => { userController.show(req, res) })
 
+router.patch('/organisation/:organisationId', (req, res) => { userController.updateOrganisation(req, res) })
```
Comment (mkayad) [inline src/routes/api.ts:51]: I think the path should be plural to be consistent with the rest of routes.

File: src/operations/updateOrganisation.ts (hunk 3 above)
Comment (Lee365) [reply src/operations/updateOrganisation.ts:3]: export interface UpdateItemInput 
    TableName: string | undefined;

**Review by mkayad — CHANGES_REQUESTED** [review]

**Pull request merged**